pkg/collector/corechecks/snmp/report_metrics.go

```diff
@@ -0,0 +1,144 @@
+package snmp
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type metricSender struct {
+	sender           aggregator.Sender
+	submittedMetrics int
+}
+
+func (ms *metricSender) reportMetrics(metrics []metricsConfig, values *resultValueStore, tags []string) {
+	for _, metric := range metrics {
+		if metric.Symbol.OID != "" {
+			ms.reportScalarMetrics(metric, values, tags)
+		} else if metric.Table.OID != "" {
+			ms.reportColumnMetrics(metric, values, tags)
+		}
+	}
+}
+
+func (ms *metricSender) getCheckInstanceMetricTags(metricTags []metricTagConfig, values *resultValueStore) []string {
+	var globalTags []string
+
+	for _, metricTag := range metricTags {
+		value, err := values.getScalarValues(metricTag.OID)
+		if err != nil {
+			log.Warnf("metric tags: error getting scalar value: %v", err)
+			continue
+		}
+		globalTags = append(globalTags, metricTag.getTags(value.toString())...)
+	}
+	return globalTags
+}
+
+func (ms *metricSender) reportScalarMetrics(metric metricsConfig, values *resultValueStore, tags []string) {
+	value, err := values.getScalarValues(metric.Symbol.OID)
```

Comment:
Good point, it's returning a single value. Renaming it to `getScalarValue`, I think `fromOID` is not necessary since it's already mentioned in the signature.

Reply:
Renamed here: https://github.com/DataDog/datadog-agent/pull/6749/commits/afb81e6a09b01b387089903916f4e825b16b42e5